keras/layers/core.py

```diff
@@ -1121,16 +1121,19 @@ def output_shape(self):
 
     def get_output(self, train=False):
         X = self.get_input(train)
-
-        def step(x, states):
-            output = K.dot(x, self.W) + self.b
-            return output, []
-
-        last_output, outputs, states = K.rnn(step, X,
-                                             initial_states=[],
-                                             mask=None)
-        outputs = self.activation(outputs)
-        return outputs
+        if K._BACKEND == 'theano':
+            import theano.tensor as T
+            z = T.tensordot(X, self.W, axes=[(2), (0)])
+            Y = z + self.b
+            Y = self.activation(Y)
+            return Y
+        elif K._BACKEND == 'tensorflow':
+            shape = self.input_shape
+            X = K.reshape(X, (-1, shape[-1]))
+            Y = K.dot(X, self.W) + self.b
+            Y = K.reshape(Y, (-1, shape[1], shape[2]))
+            Y = self.activation(Y)
+            return Y
```

Comment:
It would be bad practice to have segments of code dedicated to one backend or another. Would there be a performance impact to just using the `K`-based code? 

Or could we reimplement `tensordot` in TensorFlow? (and make it part of the backend).